kafka_consumer_freshness_tracker/src/main/java/com/tesla/data/consumer/freshness/ConsumerFreshness.java

```diff
@@ -117,6 +124,17 @@ private void setup(Map<String, Object> conf) {
     int workerThreadCount = (int) conf.getOrDefault("workerThreadCount", DEFAULT_WORKER_THREADS);
     this.availableWorkers = ((List<Map<String, Object>>) conf.get("clusters")).stream()
         .map(clusterConf -> {
+          // validate the cluster configuration
+          validateClusterConf(clusterConf)
+              .map(error -> String.format("configuration for cluster %s is invalid: %s",
+                              clusterConf.get("name"), error))
+              .ifPresent(msg -> {
+                if (strict) {
+                  throw new RuntimeException(msg);
```

Comment:
one test that covers this case - that we do actually fail to create the workers - would be good to ensure strict is doing what the README claims.

Sorry I wasn't clear before - didn't mean to you should get rid of all of them, just that as the general coverage approach it was weird. However, still worth validating that if there is a strict failure it throws an exception.